s-c-s/compromises: contributing some new stories plus a comment for the README (#324)
cncf/tag-security: supply-chain-security/compromises/README.md

We welcome additions to this catalog by 
[filing an issue](https://github.com/cncf/sig-security/issues/new/choose) or
submitting a [pull request](https://github.com/cncf/sig-security) to our repo.

Additions aren't intended to be a statement on the security stance and/or
practices of any given project, or of open source as a whole. If anything,
many of the stories showcase rapid response, evolving security practices in

> the community and an open and transparent approach to disclosure.

Reviewer: I like this banner, personally. I just wanted to add a note that I acknowledge this change and approve it. I'd like to know @lumjjb's, @TheFoxAtWork's and @ultrasaurus's opinion on it.

Reviewer 2: Love this banner.  Perhaps some rephrasing to be more clear:

> Contents of this repo and proposed additions are not a statement or opinion on the security stance and/or practices of a given project, of open source, or the community.  These articles and stories annotate the communities dedication to rapid response, evolving security practices, transparent disclosure, and enforcement of one of open sources founding principles, "[Linus's Law](https://en.wikipedia.org/wiki/Linus%27s_law)".

Reviewer 2: Also perhaps adding:

> When submitting an addition, please review the [definitions](https://github.com/cncf/sig-security/blob/master/supply-chain-security/compromises/compromise-definitions.md) page to ensure the Type of Compromise on the details of the incidents as well as the Catalog itself are consistent.  If a definition doesn't exist or a new type of compromise needs added, please include that as well.

> Additions aren't intended to be a statement on the security stance and/or
> practices of any given project, or of open source as a whole. If anything,
> many of the stories showcase rapid response, evolving security practices in
> the community and an open and transparent approach to disclosure.

Reviewer 2: ```suggestion
Contents of this repo and proposed additions are not a statement or opinion on the security stance and/or practices of a given project, of open source, or the community. These articles and stories annotate the communities dedication to rapid response, evolving security practices, transparent disclosure, and enforcement of one of open sources founding principles, "[Linus's Law](https://en.wikipedia.org/wiki/Linus%27s_law)".
When submitting an addition, please review the [definitions](https://github.com/cncf/sig-security/blob/master/supply-chain-security/compromises/compromise-definitions.md) page to ensure the Type of Compromise on the details of the incidents as well as the Catalog itself are consistent. If a definition doesn't exist or a new type of compromise needs added, please include that as well.
```

| Name              | Year               | Type of compromise    | Link        |
| ----------------- | ------------------ | ------------------    | ----------- |
| [electron-native-notify](2019/electron-native-notify.md) | 2019 | Source Code Compromise | [1](https://blog.npmjs.org/post/185397814280/plot-to-steal-cryptocurrency-foiled-by-the-npm), [2](https://komodoplatform.com/update-agama-vulnerability/)|

> | [PyPI typosquatting](2019/pypi.md) | 2019 | Naming attack | [1](https://blog.reversinglabs.com/blog/suppy-chain-malware-detecting-malware-in-package-manager-repositories) |

Reviewer 2: Change `(Naming Attack)` to `(Negligence)` for consistency with definitions

> | [PyPI typosquatting](2019/pypi.md) | 2019 | Naming attack | [1](https://blog.reversinglabs.com/blog/suppy-chain-malware-detecting-malware-in-package-manager-repositories) |

Reviewer 2: ```suggestion
| [PyPI typosquatting](2019/pypi.md) | 2019 | Negligence | [1](https://blog.reversinglabs.com/blog/suppy-chain-malware-detecting-malware-in-package-manager-repositories) |
```

> | [ROS build farm compromise](2019/ros.md) | 2019 | Build infrastructure | [1](https://discourse.ros.org/t/security-issue-on-ros-build-farm/9342/8), [2](https://discourse.ros.org/t/new-gpg-keys-deployed-for-packages-ros-org/9454) |

Reviewer 2: change `(Build infrastructure)` to `(Trust and Signing)` and add `(Publishing Infrastructure)` as a new line in the table for consistency with definitions.

> | [ROS build farm compromise](2019/ros.md) | 2019 | Build infrastructure | [1](https://discourse.ros.org/t/security-issue-on-ros-build-farm/9342/8), [2](https://discourse.ros.org/t/new-gpg-keys-deployed-for-packages-ros-org/9454) |

Reviewer 2: ```suggestion
| [ROS build farm compromise](2019/ros.md) | 2019 | Trust and Signing <br>Publishing Infrastructure</br> | [1](https://discourse.ros.org/t/security-issue-on-ros-build-farm/9342/8), [2](https://discourse.ros.org/t/new-gpg-keys-deployed-for-packages-ros-org/9454) |
```

| [ShadowHammer](2019/shadowhammer.md) | 2019 | Multiple steps | [1](https://www.csoonline.com/article/3384259/asus-users-fall-victim-to-supply-chain-attack-through-backdoored-update.html), [2](https://securelist.com/operation-shadowhammer/89992/) |
| [PEAR Breach](2019/pear.md) | 2019 | Publishing Infrastructure | [1](https://blog.dcso.de/php-pear-software-supply-chain-attack/), [2](https://thehackernews.com/2019/01/php-pear-hacked.html) |

> | [Canonical's GitHub org compromised](2019/canonical-github.md) | 2019 | Build and publishing infrastructure | [1](https://www.zdnet.com/article/canonical-github-account-hacked-ubuntu-source-code-safe/) |

Reviewer 2: Change `(build)` to `(Dev Tooling) `and add `(Source Code)` before `(Publishing Infrastructure)` as new lines in the table for consistency with definitions.

> | [Canonical's GitHub org compromised](2019/canonical-github.md) | 2019 | Build and publishing infrastructure | [1](https://www.zdnet.com/article/canonical-github-account-hacked-ubuntu-source-code-safe/) |

Reviewer 2: ```suggestion
| [Canonical's GitHub org compromised](2019/canonical-github.md) | 2019 | Dev Tooling <br>Source Code </br> <br> Publishing infrastructure </br> | [1](https://www.zdnet.com/article/canonical-github-account-hacked-ubuntu-source-code-safe/) |
```

| [Dofoil](2018/dofoil.md) | 2018 | Publishing Infrastructure | [1](https://www.zdnet.com/article/windows-attack-poisoned-bittorrent-client-set-off-huge-dofoil-outbreak-says-microsoft/) |
| [Operation Red](2018/operation-red.md) | 2018 | Publishing Infrastructure | [1](https://blog.trendmicro.com/trendlabs-security-intelligence/supply-chain-attack-operation-red-signature-targets-south-korean-organizations/) |

> | [RCE in go get -u](2018/gogetu.md) | 2018 | Tooling attack | [1](https://github.com/golang/go/issues/29230), [2](https://staaldraad.github.io/post/2019-03-28-go-get-vuln/) |

Reviewer 2: Change `(Tooling Attack)` to ('Dev Tooling)' for consistency with definitions.

> | [RCE in go get -u](2018/gogetu.md) | 2018 | Tooling attack | [1](https://github.com/golang/go/issues/29230), [2](https://staaldraad.github.io/post/2019-03-28-go-get-vuln/) |

Reviewer 2: ```suggestion
| [RCE in go get -u](2018/gogetu.md) | 2018 | Dev Tooling | [1](https://github.com/golang/go/issues/29230), [2](https://staaldraad.github.io/post/2019-03-28-go-get-vuln/) |
```

> | [acroread compromised in AUR](2018/aur.md) | 2018 | Naming attack | [1](ttps://lists.archlinux.org/pipermail/aur-general/2018-July/034152.html), [2](https://www.bleepingcomputer.com/news/security/malware-found-in-arch-linux-aur-package-repository/) |

Reviewer 2: Change `(Naming attack)` to either`(Orphaned Package Takeover)` or '(Malicious Adoption)' for consistency with previous comment regarding acroread.

Reviewer 2: @SantiagoTorres do you concur?

Reviewer 2: Malicious Maintainer - definition already added updating here.

| [Bitcoin Gold](2017/bitcoingold.md) | 2017 | Source Code Compromise | [1](https://bitcoingold.org/critical-warning-nov-26/) | 
| [ExpensiveWall](2017/expensivewall.md) | 2017 | Backdooring SDK | [1](https://blog.checkpoint.com/2017/09/14/expensivewall-dangerous-packed-malware-google-play-will-hit-wallet/), [2](https://research.checkpoint.com/expensivewall-dangerous-packed-malware-google-play-will-hit-wallet/)
| [OSX Elmedia player](2017/elmedia.md) | 2017 | Publishing infrastructure | [1](https://www.hackread.com/hackers-infect-mac-users-proton-malware-using-elmedia-player/) |

> | [GitHub password recovery issues](2016/gh-unicode.md) | 2016 | SCM | [1](https://bounty.github.com/researchers/jagracey.html), [2](https://eng.getwisdom.io/hacking-github-with-unicode-dotless-i/) |

Reviewer 2: Change SCM to `(Source Code)` and `(Dev Tooling)`

> | [GitHub password recovery issues](2016/gh-unicode.md) | 2016 | SCM | [1](https://bounty.github.com/researchers/jagracey.html), [2](https://eng.getwisdom.io/hacking-github-with-unicode-dotless-i/) |

Reviewer 2: ```suggestion
| [GitHub password recovery issues](2016/gh-unicode.md) | 2016 | Dev Tool <br> Source Code </br> | [1](https://bounty.github.com/researchers/jagracey.html), [2](https://eng.getwisdom.io/hacking-github-with-unicode-dotless-i/) |
```

| [Ceph and Inktank](2015/ceph-and-inktank.md) | 2015 | Build, source and publishing infrastructure | [1](https://www.zdnet.com/article/red-hats-ceph-and-inktank-code-repositories-were-cracked/) |
| [Code Spaces](2014/code-spaces.md) | 2014    | Source Code Compromise| [1](https://threatpost.com/hacker-puts-hosting-service-code-spaces-out-of-business/106761/)
| [Monju Incident](2014/monju.md) | 2014    | Publishing infrastructure| [1](https://www.contextis.com/en/blog/context-threat-intelligence-the-monju-incident)

> | [APT lack of validation for source packages](2013/apt.md) | 2013 | Build infrastructure | [1](https://lwn.net/Articles/602461/) |

Reviewer 2: Change `(Build infrastructure)` to `(Negligence)`

> | [APT lack of validation for source packages](2013/apt.md) | 2013 | Build infrastructure | [1](https://lwn.net/Articles/602461/) |

Reviewer 2: ```suggestion
| [APT lack of validation for source packages](2013/apt.md) | 2013 | Negligence | [1](https://lwn.net/Articles/602461/) |
```

| [kernel.org compromise](2011/kernelorg.md) | 2011 | Publishing infrastructure | [1](https://lwn.net/Articles/461237/), [2](https://lwn.net/Articles/461552/) |

> | [apache.org incident](2010/apache.md) | 2010 | Build and publishing infrastructure | [1](https://blogs.apache.org/infra/entry/apache_org_04_09_2010) |

Reviewer 2: Change `(Build and publishing infrastructure)` to `(Attack Chaining)`

> | [apache.org incident](2010/apache.md) | 2010 | Build and publishing infrastructure | [1](https://blogs.apache.org/infra/entry/apache_org_04_09_2010) |

Reviewer 2: ```suggestion
| [apache.org incident](2010/apache.md) | 2010 | Attack Chaining | [1](https://blogs.apache.org/infra/entry/apache_org_04_09_2010) |
```

| [Operation Aurora](2010/aurora.md) | 2010 | Watering-hole attack | [1](https://www.wired.com/2010/03/source-code-hacks/) | 
| [ProFTPD](2010/proftpd.md) | 2010 | Source Code Repository | [1](https://www.zdnet.com/article/open-source-proftpd-hacked-backdoor-planted-in-source-code/) |

> | [WordPress backdoor](2007/wordpress.md) | 2007 | Source Code Repository | [1](https://lwn.net/Articles/224997/) |

Reviewer 2: remove `(Repository)` for consistency with definitions and add `(Publishing Infrastructure)` as a new line

> | [WordPress backdoor](2007/wordpress.md) | 2007 | Source Code Repository | [1](https://lwn.net/Articles/224997/) |

Reviewer 2: ```suggestion
| [WordPress backdoor](2007/wordpress.md) | 2007 | Source Code <br> Publishing Infrastructure </br> | [1](https://lwn.net/Articles/224997/) |
```

> | [SquirrelMail backdoor](2007/squirrelmail.md) | 2007 | Source Code Repository | [1](https://lwn.net/Articles/262688/) |

Reviewer 2: remove `(Repository)` for consistency with definitions

> | [SquirrelMail backdoor](2007/squirrelmail.md) | 2007 | Source Code Repository | [1](https://lwn.net/Articles/262688/) |

Reviewer 2: ```suggestion
| [SquirrelMail backdoor](2007/squirrelmail.md) | 2007 | Source Code | [1](https://lwn.net/Articles/262688/) |
```
